app/helpers/form_errors_builder.rb

```diff
@@ -0,0 +1,29 @@
+class FormErrorsBuilder < ActionView::Helpers::FormBuilder
```

Comment:
I'm still not sold.
```erb
<!-- app/views/shared/_errors.html.erb -->
<% if object.respond_to(:errors) && object.errors.any? %>
  <div class="panel panel-danger">
    <div class="panel-heading">
      <h3 class="panel-title">
        Please fix the errors below and resubmit the form.
      </h3>
    </div>

    <div class="panel-body">
      <ul class="text-danger">
        <% object.errors.full_messages.each do |message| %>
          <li><%= message %></li>
        <% end %>
      </ul>
    </div>
  </div>
<% end %>

<!-- app/views/hq/<model>/_form.html.erb -->
<%= form_for [:hq, @model] do |f| %>
  <%= render 'shared/errors', object: f.object %>
  .
  .
<% end %>
```
 - the .erb file is almost half the size of the Ruby one
 - given that it's preparing (part of) a view to be rendered, the fact that the structure of that view is immediately visible seems of value
 - content tags are most useful when they interpolate CSS selectors and/or content but I see no utility in using them to generate static markup and largely static content. Overall, this applies to Ruby vs HTML - doesn't the latter feel like the right tool for the job?
 - there seems to be no difference in terms of encapsulation - the form either calls a helper or renders a partial